gluonnlp/model/char_encoder.py

```diff
@@ -0,0 +1,144 @@
+    num_filters: int
+        The output dimension for each convolutional layer,
+        which is the number of the filters learned by the layer.
+    ngram_filter_sizes: Tuple[int]
+        The size of each convolutional layer,
+        and len(ngram_filter_sizes) equals to the number of convolutional layers.
+    conv_layer_activation: nn.Activation
+        Activation function to be used after convolutional layer.
+    num_highway: int
+        The number of layers of the Highway layer.
+    output_size: int
+        The output dimension after conducting the convolutions and max pooling,
+        and apply highways, as well as linear projection.
+
+    """
+    def __init__(self,
+                 embed_size,
+                 num_filters,
+                 ngram_filter_sizes,
+                 conv_layer_activation=nn.Activation('relu'),
+                 num_highway=None,
+                 output_size=None,
+                 **kwargs):
+        super(CharacterEncoder, self).__init__(**kwargs)
+
+        self._embed_size = embed_size
+        self._num_filters = num_filters
+        self._ngram_filter_sizes = ngram_filter_sizes
+        self._num_highway = num_highway
+        self._output_size = output_size
+
+        with self.name_scope():
+            self._convs = nn.HybridSequential()
+            with self._convs.name_scope():
+                for _, ngram_size in enumerate(self._ngram_filter_sizes):
+                    self._convs.add(nn.Conv1D(in_channels=self._embed_size,
+                                              channels=self._num_filters,
+                                              kernel_size=ngram_size,
+                                              use_bias=True))
+            maxpool_output_size = self._num_filters * len(self._ngram_filter_sizes)
+            self._activation = conv_layer_activation
+            if self._num_highway:
+                self._highways = Highway(maxpool_output_size,
+                                         self._num_highway,
+                                         activation=self._activation)
+            else:
+                self._highways = None
+            if self._output_size:
+                self._projection = nn.Dense(in_units=maxpool_output_size,
+                                            units=self._output_size,
+                                            use_bias=True)
+            else:
+                self._projection = None
+                self._output_size = maxpool_output_size
+
+    def set_highway_bias(self):
+        self._highways.set_bias()
+
+    def forward(self, inputs, mask=None): # pylint: disable=arguments-differ
+        r"""
+        Forward computation for char_encoder
+
+        Parameters
+        ----------
+        inputs: NDArray
+            The input tensor is of shape `(seq_len, batch_size, embedding_size)` TNC.
+        mask: NDArray
+            The mask applied to the input of shape `(seq_len, embedding_size)`
```

Comment:
Why is mask 2-D?

Reply:
because we are using ``inputs * mask.expand_dims(-1)`` , the mask will be broadcasted along the embedding dimension.

Reply:
Yes it is a mistake, thanks. Updated